consumers/jira_c/gojira/gojira.go

```diff
@@ -0,0 +1,88 @@
+package gojira
+
+import (
+	"io/ioutil"
+	"log"
+
+	"github.com/andygrunwald/go-jira"
+	"github.com/trivago/tgo/tcontainer"
+)
+
+type goJiraClient struct {
+	JiraClient *jira.Client
+	DryRunMode bool
+	Configs    config
+}
+
+// NewGoJiraClient returns a goJiraClient containing the authentication details and the configuration settings
+func NewGoJiraClient(user, token, url string, dryRun bool) *goJiraClient {
+	return &goJiraClient{
+		JiraClient: authJiraClient(user, token, url),
+		DryRunMode: dryRun,
+		Configs:    getConfig(),
+	}
+}
+
+// authJiraClient authenticates the client with the given Username, API token, and URL domain
+func authJiraClient(user, token, url string) *jira.Client {
+	tp := jira.BasicAuthTransport{
+		Username: user,
+		Password: token,
+	}
+	jiraClient, err := jira.NewClient(tp.Client(), url)
+	if err != nil {
+		log.Fatalf("Unable to contact Jira: %s", err)
+	}
+	return jiraClient
+}
+
+// assembleIssue parses the Dracon message and serializes it into a Jira Issue object
+func (client goJiraClient) assembleIssue(message string) *jira.Issue {
+	// Assign the default CustomField values specified in the configuration
+	customFields := tcontainer.NewMarshalMap()
+	for _, cf := range client.Configs.DefaultValues.CustomFields {
```

Comment:
Run this code only once, after the config.yaml is parsed; right now it's being run everytime a new issue is created

Reply:
Created a `defaultFields` variable that contains all these default fields specified by the user. They only get created once and the code can just reference their values (or deep-copy them in the case of CustomFields).